sql/core/src/test/scala/org/apache/spark/sql/sources/PartitionedWriteSuite.scala

```diff
@@ -156,4 +156,13 @@ class PartitionedWriteSuite extends QueryTest with SharedSparkSession {
       }
     }
   }
+
+  test("SPARK-31968:duplicate partition columns check") {
+    val ds = Seq((3, 2)).toDF("a", "b")
+    val e = intercept[AnalysisException](ds
+      .write.mode(org.apache.spark.sql.SaveMode.Overwrite)
```

Comment:
I suppose it doesn't have to be overwrite mode. Let's also make it inlined properly within 100 line length limit. See also https://github.com/databricks/scala-style-guide